src/librustc_mir/hair/cx/expr.rs

```diff
@@ -596,8 +597,18 @@ fn method_callee<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
         span: expr.span,
         kind: ExprKind::Literal {
             literal: Literal::Value {
-                value: cx.tcx.mk_const(ty::Const {
-                    val: ConstVal::Function(def_id, substs),
+                value: cx.tcx().mk_const(ty::Const {
+                    val: if cx.tcx().sess.opts.debugging_opts.miri {
+                        let inst = ty::Instance::new(def_id, substs);
+                        let ptr = cx.tcx()
```

Comment:
@eddyb help! I'm not sure where to start touching the hair in order to get these 'gcx and 'tcx lifetimes satisfied:

```
error[E0623]: lifetime mismatch
   --> src/librustc_mir/hair/cx/expr.rs:603:38
    |
584 | fn method_callee<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
    |                                           ------------------ this parameter and the return type are declared with different lifetimes...
...
587 |                                  -> Expr<'tcx> {
    |                                     ----------
...
603 |                         let ptr = cx.tcx()
    |                                      ^^^ ...but data from `cx` is returned here

error: aborting due to previous error

error: Could not compile `rustc_mir`.
```